core/src/main/java/org/elasticsearch/action/support/DestructiveOperations.java

```diff
@@ -21,25 +21,30 @@
 
 import org.elasticsearch.common.component.AbstractComponent;
 import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.node.settings.NodeSettingsService;
+import org.elasticsearch.common.settings.ClusterSettingsService;
 
 /**
  * Helper for dealing with destructive operations and wildcard usage.
  */
-public final class DestructiveOperations extends AbstractComponent implements NodeSettingsService.Listener {
+public final class DestructiveOperations extends AbstractComponent {
 
     /**
      * Setting which controls whether wildcard usage (*, prefix*, _all) is allowed.
      */
-    public static final String REQUIRES_NAME = "action.destructive_requires_name";
+    public static final Setting<Boolean> REQUIRES_NAME_SETTING = Setting.boolSetting("action.destructive_requires_name", false, true, Setting.Scope.Cluster);
```

Comment:
Builder pattern may make this more clear.
```
Setting<Boolean> REQUIRES_NAME_SETTING = Setting.builder("action.destructive_requires_name").dynamic().clusterScope().defaultsTo(false);
```

Right now you have to remember the canonical parameter order for these builder methods. And it'd be easier to have default values with the builder implementation I think.